maven-model-builder/src/main/java/org/apache/maven/model/interpolation/StringSearchModelInterpolator.java

```diff
@@ -250,7 +250,13 @@ private static void evaluateArray( Object target, InterpolateObjectAction ctx )
 
             private boolean isQualifiedForInterpolation( Class<?> cls )
             {
-                return !cls.getName().startsWith( "java" );
+                Package pkg = cls.getPackage();
```

Comment:
As you have the Class object.
Will it make sense to exclude every system class, the ones loaded from the system classloader (class.getClassloader().getParent() == null) ?

Reply:
@eolivelli 
Hi Enrico,
System classes were excluded before this change as well in another place.
The purpose of this class is to interpolate Maven expressions and resolve them to `String`.

Given objects which are passed into these routines can be complex objects with delegates and inheritance as well. This algorithm is traversing all the hierarchy of given object, and the leaf of this structure is anyway supposed to be the String. 

Finally, the string with the expression is overridden by interpolated string. This works fine without my change.

My change only guarantees that the internal JDK fields of e.g. `java.io.File` which are not Maven expressions are not introspected further. Anyway it does not make sense to continue introspecting them further.
Reading the internal fields of e.g. `File` is what Java 12 does not like, Java prints warnings and so we should rather avoid reading them.

There are much more troubles with this class. As for instance we are also reading `static` fields. There are warnings shown by IDEA saying that two instance variables can be converted to local variables, etc. I would like to post a new PR for these improvements.